services/tinlake/actions.ts

```
@@ -86,17 +86,21 @@ export const mintNFT = async (
   ref: string,
   amount: string,
   asset: string
-) => {
+): Promise<PendingTransaction> => {
   try {
     return await tinlake.mintNFT(nftAddr, owner, tokenId, ref, amount, asset)
   } catch (e) {
     return loggedError(e, 'Could not mint NFT.', tokenId)
   }
 }
 
-export const issue = async (tinlake: ITinlake, tokenId: string, nftRegistryAddress: string): Promise<TinlakeResult> => {
+export const issue = async (
+  tinlake: ITinlake,
+  tokenId: string,
+  nftRegistryAddress: string
+): Promise<PendingTransaction> => {
   let tokenOwner
-  const user = (tinlake.ethConfig as any).from!
+  const user = await tinlake.signer?.getAddress()!
```

Should we fail here if user is not set?